src/NCronJob/Registry/JobRegistry.cs

```diff
@@ -28,13 +28,13 @@ public void Add(JobDefinition jobDefinition)
     {
         AssertNoDuplicateJobNames(jobDefinition.CustomName);
 
-        var isTypeUpdate = allJobs.Any(j => j.JobFullName == jobDefinition.JobFullName);
-        if (isTypeUpdate)
+        if (!allJobs.Add(jobDefinition))
         {
-            Remove(jobDefinition);
+            throw new InvalidOperationException("""
+                                                A duplicated job registration was detected (the same job type with the same parameter and cron expression and not custom name).
```

Comment:
Couldn't we provide in the exception message some info about the jobdefinition (type/name/cron when they exist)?
That would make it easier for the user to quickly figure out what to search for.

Reply:
The type at least would be something. I will add this tomorrow. Thanks!

Reply:
And maybe refine the error message in `AssertNoDuplicateJobNames` so that it's worded in the same manner ("A duplicated job registration was detected, ...")?

Reply:
@linkdotnet So no ❤️ for better messages? :wink:

Reply:
Well there is but I will now use ChatGPT for the better :D

Reply:
Soooooo. I hope that is a bit better

Reply:
Duh. I was unclear in my previous message.

I was referring to maybe add the duplicate type in the error message and maybe reword what AssertNoDuplicateJobNames throws.

Reply:
Ah darn me. Sorry. Will do